libraries/CurieBLE/examples/LEDCentral/LEDCentral.ino

```diff
@@ -0,0 +1,161 @@
+    blePeripheral1->discover();
+}
+
+void bleCentralDisconnectHandler(BLEHelper& peripheral)
+{
+    // peripheral disconnected event handler
+    blePeripheral1 = NULL;
+    Serial.print("Disconnected event, peripheral: ");
+    Serial.println(peripheral.address());
+    bleCentral.startScan();
+}
+
+void switchCharacteristicWritten(BLEHelper& peripheral, BLECharacteristic& characteristic)
+{
+    // Read response/Notification wrote new value to characteristic, update LED
+    Serial.print("Characteristic event, written: ");
+
+    if (switchChar.value()) 
+    {
+        Serial.println("LED on");
+        digitalWrite(ledPin, HIGH);
+    }
+    else
+    {
+        Serial.println("LED off");
+        digitalWrite(ledPin, LOW);
+    }
+}
+
+bool adv_found(uint8_t type, 
+               const uint8_t *data, 
+               uint8_t data_len,
+               void *user_data)
+{
+    bt_addr_le_t *addr = (bt_addr_le_t *)user_data;
+    int i;
+
+    Serial.print("[AD]:");
+    Serial.print(type);
+    Serial.print(" data_len ");
+    Serial.println(data_len);
+
+    switch (type)
+    {
+        case BT_DATA_UUID128_SOME:
+        case BT_DATA_UUID128_ALL:
+        {
+            if (data_len % MAX_UUID_SIZE != 0) 
+            {
+                Serial.println("AD malformed");
+                return true;
+            }
+            struct bt_uuid * serviceuuid = ledService.uuid();
+            for (i = 0; i < data_len; i += MAX_UUID_SIZE)
+            {
+                if (memcmp (((struct bt_uuid_128*)serviceuuid)->val, &data[i], MAX_UUID_SIZE) != 0) 
+                {
+                    continue;
+                }
+                
+                // Accept the advertisement
+                if (!bleCentral.stopScan()) 
+                {
+                    Serial.println("Stop LE scan failed");
+                    continue;
+                }
+                Serial.println("Connecting");
+                // Connect to peripheral
+                bleCentral.connect(addr, &conn_param);
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+void setup() {
+    Serial1.begin(115200);
+	
+    Serial.begin(9600);
+    pinMode(ledPin, OUTPUT); // use the LED on pin 13 as an output
+
+    // add service and characteristic
+    bleCentral.addAttribute(ledService);
+    bleCentral.addAttribute(switchChar);
+
+    // assign event handlers for connected, disconnected to central
+    bleCentral.setEventHandler(BLEConnected, bleCentralConnectHandler);
+    bleCentral.setEventHandler(BLEDisconnected, bleCentralDisconnectHandler);
+
+    // advertise the service
+    bleCentral.setAdvertiseHandler(adv_found);
+
+    // assign event handlers for characteristic
+    switchChar.setEventHandler(BLEWritten, switchCharacteristicWritten);
+
+    bleCentral.begin();
+    Serial.println(("Bluetooth device active, waiting for connections..."));
+}
+
+void loop()
+{
+    static unsigned char counter = 0;
```

Comment:
Why do you need `static unsigned char` and `static char` for these? `counter` is ... well, just a counter, and `ledstate` only flips back and forth between 0 and 1. To avoid confusing users, maybe use `int` or `short` for both?